gulpfile.js

```diff
@@ -51,6 +52,49 @@ const { launch, connect } = require('hadouken-js-adapter');
 		console.log(`[${new Date().toLocaleTimeString()}] ${chalk[color][bgcolor](msg)}.`);
 	}
 
+	/** 
+	* Validates the current node version against supported node versions specified in this file
+	* Returns true of false if current node version is valid
+	* Currently only validates against a max node version
+	*
+	* Note: This method is being used instead of npm engines because of an npm bug where warnings don't print
+	* This bug was resolved in npm 6.12.0 but as that is a very new version of npm and is not linked to node 10.15.3
+	* in nvm we can't assume our users have access to this version.
+	*/
+	const isNodeVersionValid = () => {
+		let ret = true;
+		// split the current node version into major, minor and patch numbers
+		let version = process.version.split('v');
+		version = version[1].split('.');		
+		const majorCurrent = Number(version[0]);
+		const minorCurrent = Number(version[1]);
+		const patchCurrent = Number(version[2]);
+
+		// split the max node version into major, minor and patch numbers
+		let maxNodeArray = MAX_NODE_VERSION.split('.');
+		if (maxNodeArray.length !== 3) {
+			console.log("Format of maximum node version must be: 'X.X.X', unable to validate node version");
+			return ret;
```

Comment:
It looks to me like this will return `true` if we are unable to validate the node version; is that what we want?

Reply:
Also, maybe we could make `const MAX_NODE_VERSION_ARRAY = [10,15,3]` so that we don't have to parse the constant string?

Reply:
It will return true if we don't define a max node version correctly in this file, which would just mean someone updated it incorrectly or removed it not that the node version we're using is bad. I figured for a configuration error we shouldn't report that the node version was wrong. If we returned false here it would log too errors which may be confusing. We can discuss a better way to handle this, ideally we could trust whoever edits this file to not mess up this value, but I wanted some type of protection.

If we started with an array it would remove one step parsing the max node version, but I figured the extra step was worth it to make the max node version more user readable.